src/Orc.FileSystem/Services/Interfaces/IFileService.cs

```diff
@@ -4,12 +4,12 @@
 
     public interface IFileService
     {
-        FileStream Create(string fileName);
+        Stream Create(string fileName);
```

Comment:
this is a breaking change
instead of modifying existing interface I would suggest to create another one IFile and for consistency I would've changed IDirectoryService to IDirectory. With the implementations FileWrapper and DirectoryWrapper.
all the old interfaces should be marked as obsolete

And for testing purposes I would recommend to implement InMemoryFile and InMemoryDirectory   

And we definitely need tests for InMemory implementations

Reply:
That's why this is done on the 5.x alpha.

We want to go through the pain as fast as possible, where this "only" requires a recompilation. I have considered introducing IFileStream, but in such case we would need to wrap the standard FileStream in a wrapper (which I think is not really needed here?).

Reply:
it will affect all the libraries which are using Orc.FileSystem indirectly.
So it is easy to reproduce like this

A -> B -> C -> D -> Orc.FileSystem
A -> E -> F -> Orc.FileSystem

, where A, B, C, ... are related libraries
if you update Orc.FileSystem in one of them and do not update in another, you will get a runtime error

So, just incrementing a major version number is not a good excuse to make breaking changes.
And when you say we want to go through the pain I cannot agree with that. I insist on making it the right way. You can change the version number to 5.x, but please make it painless